src/util/fs.h

```diff
@@ -54,10 +56,8 @@ class path : public std::filesystem::path
 
     std::string u8string() const
     {
-        const auto& utf8_str{std::filesystem::path::u8string()};
-        // utf8_str might either be std::string (C++17) or std::u8string
-        // (C++20). Convert both to std::string. This method can be removed
-        // after switching to C++20.
+        const std::u8string& utf8_str{std::filesystem::path::u8string()};
+        // Convert to std::string as a convenience for use in RPC code.
         return std::string{utf8_str.begin(), utf8_str.end()};
     }
```

Comment:
What about the comment that says "This method can be removed after switching to C++20"? I think indeed this method should be removed entirely.

In C++17 its prototype, according to the spec is `std::string u8string() const` which matches ours and is ok. But in C++20 it is: `std::u8string u8string() const` and it is confusing to have a mismatch. I would better drop this method like the comment says and then fix problems (if any) that arise due to that.

Reply:
I need to understand this better. What I grasp now is that in C++17 it was ok to store UTF-8 strings in `std::string` and in C++20 it is not and one has to use `std::u8string` instead?

This smells because if callers need `std::string` why do they call `u8string()` and not `string()`!? Is it because it was ok to do that in C++17 because in C++17 `std::filesystem::path::string()` would return `std::string` itself and on top of this it was needed because of some complication on Windows?

On Windows `filesystem::path` uses `wchar_t` and it has to be somehow converted to `std::string<char>`, I guess:
1. `filesystem::path::string()` which does that conversion is not ok
2. `filesystem::path::u8string()` converts `wchar_t` to
  2.1. C++17: `char` from which we convert it explicitly to `char` which is a noop
  2.2. C++20: `char8_t` from which we convert it explicitly to `char`. Is that ok? Why C++20 has `std::u8string` if we can store UTF-8 strings in `std::string`?

What would be an example in C++20 where it is broken by calling `string()` (1.) and ok by converting `wchar_t`->`char8_t`->`char` (2.2)?

Reply:
> This has nothing to do with C++20

ok, rephrase (I mentioned C++20 because C++17 does not have `char8_t`):

What would be an example where it is broken by calling `string()` (1.) and ok by converting `wchar_t`->`char8_t`->`char` (2.2)?

Reply:
> confusing to have a u8string method that doesn't actually return a u8string

My concern exactly!

Reply:
Or replace `UniValue` with https://github.com/nlohmann/json